transport/http/codec.go

```diff
@@ -59,10 +59,6 @@ func DefaultRequestQuery(r *http.Request, v any) error {
 
 // DefaultRequestDecoder decodes the request body to object.
 func DefaultRequestDecoder(r *http.Request, v any) error {
-	codec, ok := CodecForRequest(r, "Content-Type")
-	if !ok {
-		return errors.BadRequest("CODEC", fmt.Sprintf("unregister Content-Type: %s", r.Header.Get("Content-Type")))
-	}
 	data, err := io.ReadAll(r.Body)
```

Comment:
https://github.com/go-kratos/kratos/blob/0d5c1900601cc9a39fc0937acc7496df03d89726/transport/http/codec.go#L67C2-L69C3

need to check the error.